cdap-ui/app/hydrator/controllers/create/toppanel-ctrl.js

```diff
@@ -294,10 +304,36 @@ class HydratorPlusPlusTopPanelCtrl {
     seconds = seconds < 10 ? '0' + seconds : seconds;
     minutes = minutes < 10 ? '0' + minutes : minutes;
 
+    this.setDisplayDuration(minutes, seconds);
+  }
+
+  setDisplayDuration(minutes, seconds) {
     this.displayDuration = {
-      minutes: minutes,
-      seconds: seconds
-    };
+      minutes: !minutes ? '--' : minutes,
+      seconds: !seconds ? '--' : seconds,
+    }
+  }
+
+  updateTimerLabelAndTitle(res) {
+    // set default
+    if (!res) {
+      this.timerLabel = this.GLOBALS.en.hydrator.preview.timerLabels.DURATION;
+      this.queueStatus = '';
+      return;
+    }
+
+    const { WAITING, ACQUIRED, INIT, RUNNING } = window.CaskCommon.PREVIEW_STATUS;
+    if (res.status === WAITING && res.positionInWaitingQueue > 0) {
+      const runsAheadInQueue = res.positionInWaitingQueue;
+      this.queueStatus = `${runsAheadInQueue} ${runsAheadInQueue === 1? 'run' : 'runs'} ahead in queue`;
+      this.timerLabel = `${runsAheadInQueue} ${this.GLOBALS.en.hydrator.preview.timerLabels.PENDING}`;
+    } else if ([ WAITING, ACQUIRED, INIT, RUNNING ].includes(res.status) && this.loadingLabel !== 'Stopping') {
+      this.timerLabel = this.GLOBALS.en.hydrator.preview.timerLabels.RUNNING;
+      this.queueStatus = '';
+    } else {
+    this.timerLabel = this.GLOBALS.en.hydrator.preview.timerLabels.DURATION;
```

Comment:
nit: the indentation seems a bit off here? I'm not sure if it's because of github UI!

Reply:
good catch, fixed!